pixi.toml

```diff
@@ -0,0 +1,378 @@
+test-verbose-cpu = { depends-on = [{ task = "test-verbose-base", environment = "dev" }], description = "Run tests with verbose output (CPU)" }
+test-verbose-gpu = { depends-on = [{ task = "test-verbose-base", environment = "dev-gpu" }], description = "Run tests with verbose output (GPU)" }
+
+# Tests without GUI (suitable for CI)
+test-nogui-base = { cmd = "python -m pytest -n auto --maxprocesses=4 --dist loadgroup", env = { QT_QPA_PLATFORM = "offscreen" }, description = "Base task to run tests without GUI (suitable for CI - used by other tasks)" }
+test-nogui-cpu = { depends-on = [{ task = "test-nogui-base", environment = "dev" }], description = "Run tests without GUI (suitable for CI) with CPU" }
+test-nogui-gpu = { depends-on = [{ task = "test-nogui-base", environment = "dev-gpu" }], description = "Run tests without GUI (suitable for CI) with GPU" }
+
+# Tests without GUI and with verbose output (suitable for CI)
+test-nogui-verbose = { cmd = "python -m pytest -vs -o log_cli=true -n auto --maxprocesses=4 --dist loadgroup", env = { QT_QPA_PLATFORM = "offscreen" }, description = "Run tests without GUI (suitable for CI) with verbose output" }
+test-nogui-verbose-cpu = { depends-on = [{ task = "test-nogui-verbose", environment = "dev" }], description = "Run tests without GUI (suitable for CI) with verbose output (CPU)" }
+test-nogui-verbose-gpu = { depends-on = [{ task = "test-nogui-verbose", environment = "dev-gpu" }], description = "Run tests without GUI (suitable for CI) with verbose output (GPU)" }
+
+# PyInstaller packaging tasks
+package-pyinstaller-base = { cmd = "python PackageWithPyInstaller.py", cwd = "packaging", description = "Build PyInstaller package" }
+package-pyinstaller-cpu = { depends-on = [{ task = "package-pyinstaller-base", environment = "dev" }], description = "Build PyInstaller package (CPU)" }
+package-pyinstaller-gpu = { depends-on = [{ task = "package-pyinstaller-base", environment = "dev-gpu" }], description = "Build PyInstaller package (GPU)" }
+
+# Documentation
+build-docs = { cmd = "sphinx-apidoc -f -M -e -T -d 3 mantidimaging '**/test' '**/test_helpers' '**/eyes_tests' -o docs/api/ && sphinx-build -b html docs/ docs/_build/html", cwd = "." }
+release-notes = "python setup.py release_notes"
+
+# Combined linting task
+[feature.lint.tasks]
+# Base linting tasks (run in lint environment)
+mypy-base = { cmd = "python -m mypy --ignore-missing-imports $SOURCE_DIRS || true", description = "Run mypy type checks" }
+pyright-base = { cmd = "python -m pyright", description = "Run pyright type checks" }
+yapf-base = { cmd = "python -m yapf --parallel --diff --recursive $SOURCE_DIRS", description = "Check code formatting with yapf" }
+yapf-apply-base = { cmd = "python -m yapf -i --parallel --recursive $SOURCE_DIRS", description = "Apply yapf formatting" }
+ruff-base = { cmd = "ruff check $SOURCE_DIRS || true", description = "Run ruff linter" }
```

Comment:
Do we need the `|| true`, it would hide any failures.

Reply:
I think we should default to stopping on the first failure. That's what the makefile did and what we'll need when we call these in CI.

We could add a `nofail` version later, so that it is explicit that should not be used as a test.